### ISSUE TITLE
tikv does not handle ambiguous time zones correctly

### ISSUE THREAD

**Issue by morgo** [tikv/tikv#3721]
## Bug Report

**What version of Rust are you using?**

N/A.  Using a binary.

**What operating system and CPU are you using?**

Debian / Intel

**What did you do?**

Using the [bikeshare sample database](https://pingcap.com/docs/bikeshare-example-database/) for TiDB:
```
mysql> delete from trips where start_date < '2017-01-01 00:00:00';
ERROR 1105 (HY000): other error: unknown error StringError("[src/coprocessor/dag/executor/mod.rs:239]: unknown error StringError(\"[src/coprocessor/codec/mysql/time/mod.rs:95]: \\\'2010-11-7 1:0:8.000000000\\\' is not a valid datetime\")")
```

**What did you expect to see?**

Success.

**What did you see instead?**

In https://github.com/pingcap/tidb/issues/7768 @breeswish confirmed that this issue is related to time zones.  I see it as a common issue, since anyone with a system time that has DST could experience it at some point.

**Comment by breeswish** [comment]
Thanks a lot for reporting. This is a known issue and we don't have a near plan to fix it. Let's keep this issue open so that others will know.

**Comment by morgo** [comment]
Thanks @breeswish.  For TiDB, we will have to consider how to document it.  It might be worth putting it on the [MySQL Compatibility](https://www.pingcap.com/docs/sql/mysql-compatibility/) page.

**Comment by breeswish** [comment]
@morgo Right. I planned to did it but forgot it 🤣 Thanks a lot for your issue!
